api/app/utils/review.inference.js

```diff
@@ -77,35 +84,38 @@ async function calculateWordDiff(data) {
   );
   const totalWords = originalTextWithoutMarkdown.split(/\s+/).length;
 
+
   fieldsToConsider.forEach((field) => {
     let originalText = data.originalContent[field] || "";
     let modifiedText = data.content[field] || "";
 
     // If the field is an array, we need to convert it to a string
     // before we can calculate the diff.
     // TODO! This is a hacky solution. We should fix the transcript json format from tstbtc.
+    if (Array.isArray(originalText)) {
+      originalText = originalText.join(" ");
+    }
+
     if (Array.isArray(modifiedText)) {
       modifiedText = modifiedText.join(" ");
     }
-    if (
-      (originalText.toString().trim().startsWith("[") &&
-        originalText.toString().trim().endsWith("]")) ||
-      (modifiedText.toString().trim().startsWith("[") && modifiedText.toString().trim().endsWith("]"))
-    ) {
-      let getOriginalText = originalText.toString().trim().slice(1, -1);
-      let getModifiedText = modifiedText.trim().slice(1, -1);
-      originalText = getOriginalText
-        .split(",")
-        .map((text) => text.trim().slice(1, -1))
-        .join(" ");
-      modifiedText = getModifiedText
-        .split(",")
-        .map((text) => text.trim().slice(1, -1))
-        .join(" ");
+
+    // Check if the text contains array brackets
+    const hasOriginalArray = /\[.*?\]/.test(originalText);
+    const hasModifiedArray = /\[.*?\]/.test(modifiedText);
+
+    if (hasOriginalArray) {
+      originalText = removeArrayBrackets(originalText);
+    }
+
+    if (hasModifiedArray) {
+      modifiedText = removeArrayBrackets(modifiedText);
```

Comment:
I like the approach. Have you tested this?

Reply:
Yes